test/e2e/privilegesmgmt/ssr.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2021 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package privilegesmgmt
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	waitutil "k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	. "github.com/vmware-tanzu/velero/test/e2e"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/k8s"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/velero"
+)
+
+func SSRTest() {
+	testNS := "ssr-test"
+	client, err := NewTestClient()
+	if err != nil {
+		println(err.Error())
+	}
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for backup tests")
+
+	BeforeEach(func() {
+		var err error
+		flag.Parse()
+		Expect(err).To(Succeed())
+		if VeleroCfg.InstallVelero {
+			Expect(VeleroInstall(context.Background(), &VeleroCfg, "", false)).To(Succeed())
+		}
+	})
+
+	AfterEach(func() {
+		if VeleroCfg.InstallVelero {
+			Expect(VeleroUninstall(context.Background(), VeleroCfg.VeleroCLI, VeleroCfg.VeleroNamespace)).To(Succeed())
+		}
+		DeleteNamespace(context.TODO(), client, testNS, false)
+	})
+
+	It(fmt.Sprintf("Should create an ssr object in the %s namespace and later removed by controller", VeleroCfg.VeleroNamespace), func() {
+		ctx, _ := context.WithTimeout(context.Background(), time.Duration(time.Minute*10))
+		By(fmt.Sprintf("Create %s namespace", testNS))
+		Expect(CreateNamespace(ctx, client, testNS)).To(Succeed(),
+			fmt.Sprintf("Failed to create %s namespace", testNS))
+
+		By(fmt.Sprintf("Get version in %s namespace", testNS))
+		cmd := []string{"version", "-n", testNS}
+		Expect(VeleroCmdExec(context.Background(), VeleroCfg.VeleroCLI, cmd)).To(Succeed(),
+			fmt.Sprintf("Failed to create an ssr object in the %s namespace", testNS))
+
+		By(fmt.Sprintf("Get version in %s namespace", VeleroCfg.VeleroNamespace))
+		cmd = []string{"version", "-n", VeleroCfg.VeleroNamespace}
+		Expect(VeleroCmdExec(context.Background(), VeleroCfg.VeleroCLI, cmd)).To(Succeed(),
+			fmt.Sprintf("Failed to create an ssr object in %s namespace", VeleroCfg.VeleroNamespace))
+
+		var req *v1.ServerStatusRequestList
+		By(fmt.Sprintf("Check ssr object in %s namespace", VeleroCfg.VeleroNamespace))
```

Comment:
Seems the case doesn't check the CR content of the SSR?

Reply:
OK